### DISCUSSION TITLE
Expect fingerprint bit to be set to 1 but for certain molecules it is set to 0

### DISCUSSION THREAD

**Discussion by JackS7806** [rdkit/rdkit#4870] [Q&A] [answered]
Hello, I am highlighting bits present in a series of molecules with both features set to true and false. I notice that a feature bit and non-feature bit are shown to be the same when I highlight them. For example:

For 
![image](https://user-images.githubusercontent.com/44094161/148783354-1396288a-2720-419b-90ed-8d0ff64b170f.png)

Running
bi_ecfp = {}
fp_ecfp = rdMolDescriptors.GetMorganFingerprintAsBitVect(n, radius=2, bitInfo=bi_ecfp,useFeatures=False, nBits = 2048)
bi_fcfp = {}
fp_fcfp = rdMolDescriptors.GetMorganFingerprintAsBitVect(n, radius=2, bitInfo=bi_fcfp,useFeatures=True, nBits = 2048)
Draw.DrawMorganBit(n,507,bi_ecfp, useSVG=True)
Draw.DrawMorganBit(n,1336,bi_fcfp, useSVG=True)

Both give 
![image](https://user-images.githubusercontent.com/44094161/148783533-8d02e453-c3f9-45f3-addb-a13ae7addc17.png)
as the fingerprint bit and when I calculate it on other molecules it successfully shows the same environment for both bits. 

However, for:

o = Chem.MolFromSmiles('O=C(N[C@@H](C(C)C)C(NCC(O)=O)=O)COC1=C(Br)C2=CC=CC=C2C=C1')

![image](https://user-images.githubusercontent.com/44094161/148783664-d5f749b9-85aa-4d8a-b014-30d5b5d8a046.png)

Running the above DrawMorganBit shows the same bit for bit 507 but fails for 1336.

I am just wondering why it would show as the same bit on some molecules and not others? 

rdkit version = 2021.09.3

**Comment by dehaenw** [comment, marked as answer]
Hello @JackS7806 ,
it seems ilke the substructure you are looking at are subtly different. This is because the FCFP invariant for a COOH carbon is different than for an amide carbon, 32 instead of 0. Here is a small script that will show the radius 0 invariants for each atom to make this more clear. The fragment you are looking for probably corresponds to bit 71 instead. Indeed it is present in both structures.
```python
import rdkit
from rdkit import Chem
from rdkit.Chem import AllChem,Draw


def mol_with_invariant_label(mol,bi):
    ivdict = {}
    for key in bi:
        for tup in bi[key]:
            if tup[1] == 0:
                ivdict[tup[0]]=key
    for atom in mol.GetAtoms():
        atom.SetProp("atomLabel", str(ivdict[atom.GetIdx()]))

n = Chem.MolFromSmiles('OC(=O)[C@H](C(C)C)NC(=O)[C@H](C(C)C)NC(=O)[C@H](C(C)C)NC(=O)COc1cc(cccc2)c2cc1')
o = Chem.MolFromSmiles('OC(=O)CNC(=O)[C@H](C(C)C)NC(=O)COc1c(Br)c(cccc2)c2cc1')
for mol in [n,o]:
    bi_fcfp = {}
    fp_fcfp = AllChem.GetMorganFingerprintAsBitVect(mol,radius=2,bitInfo=bi_fcfp,useFeatures=True,nBits=2048)
    mol_with_invariant_label(mol,bi_fcfp)
    display(mol)
    display(Draw.DrawMorganBit(mol,71,bi_fcfp))
    try:
        display(Draw.DrawMorganBit(mol,1336,bi_fcfp))
    except:
        print("bit is not in fingerprint!")
```
this gives the following output:
molecule n with radius 0 invariant labels:
![image](https://user-images.githubusercontent.com/66372095/148843325-d88f47eb-5b67-4bfb-8d62-de9187745c03.png)
molecule o with radius 0 invariant labels:
![image](https://user-images.githubusercontent.com/66372095/148843379-6bfe2aad-781c-461c-9f8f-1285c95ebe02.png)
fragment that corresponds to bit 1336 (only present in molecule N):
![image](https://user-images.githubusercontent.com/66372095/148843467-4d5a94eb-43c5-4710-8f45-449549065d5d.png)
fragment that corresponds to bit 71 (present in both molecules):
![image](https://user-images.githubusercontent.com/66372095/148843570-8298ac85-0e9e-4999-8cd7-c4cc2317c26e.png)
I hope this clarifies the lack of the fragment.

**Reply by JackS7806** [reply]
Many thanks for clearing this up for me - it's quite a subtle distinction. I suppose the highlighting of the bit caused the bulk of my confusion as I was only considering the carbon* as responsible for the bit rather than the local environment

**Reply by greglandrum** [reply]
The highlighiting is quite condensed: it actually only shows you the atoms involved in defining the bit.
